packages/python/plotly/_plotly_utils/utils.py

```diff
@@ -40,10 +40,15 @@ def encode(self, o):
         Note that setting invalid separators will cause a failure at this step.
 
         """
-
         # this will raise errors in a normal-expected way
+        self.hasinfnans = False
         encoded_o = super(PlotlyJSONEncoder, self).encode(o)
-
+        # Brute force guessing whether NaN or Infinity values are in the string
+        # We catch false positive cases (e.g. strings such as titles, labels etc.)
+        # but this is ok since the intention is to skip the decoding / reencoding
+        # step when it's completely safe
+        if not ("Infinity" in encoded_o or "NaN" in encoded_o):
+            return encoded_o
```

Comment:
I'm fine with this for now given your description of the performance characteristics.

Reply:
maybe check for `NaN` first, as it feels like that'd appear more frequently?